quinn-udp/src/unix.rs

```diff
@@ -725,6 +746,13 @@ fn decode_recv(
             (libc::SOL_UDP, gro::UDP_GRO) => unsafe {
                 stride = cmsg::decode::<libc::c_int, libc::cmsghdr>(cmsg) as usize;
             },
+            #[cfg(target_os = "linux")]
+            (libc::SOL_SOCKET, libc::SCM_TIMESTAMP) => {
+                let timeval = unsafe { cmsg::decode::<libc::timeval, libc::cmsghdr>(cmsg) };
+                let secs = Duration::from_secs(timeval.tv_sec as u64);
+                let usecs = Duration::from_micros(timeval.tv_usec as u64);
```

Comment:
Feels like [`Duration::new()`](https://doc.rust-lang.org/std/time/struct.Duration.html#method.new) might be more idiomatic for this?